config/v1alpha1/types_insights.go

```diff
@@ -48,24 +47,78 @@ type GatherConfig struct {
 	// When set to None the data is not obfuscated.
 	// When set to ObfuscateNetworking the IP addresses and the cluster domain name are obfuscated.
 	// When omitted, this means no opinion and the platform is left to choose a reasonable default, which is subject to change over time.
-	// The current default is None.
 	// +optional
 	DataPolicy DataPolicy `json:"dataPolicy,omitempty"`
 	// disabledGatherers is a list of gatherers to be excluded from the gathering. All the gatherers can be disabled by providing "all" value.
 	// If all the gatherers are disabled, the Insights operator does not gather any data.
+	// The format for the disabledGatherer should be: {gatherer}/{function} where the function is optional.
+	// Gatherer consists of a lowercase string that may include underscores (_).
+	// Function consists of a lowercase string that may include underscores (_) and is separated from the gatherer by a forward slash (/).
 	// The particular gatherers IDs can be found at https://github.com/openshift/insights-operator/blob/master/docs/gathered-data.md.
 	// Run the following command to get the names of last active gatherers:
 	// "oc get insightsoperators.operator.openshift.io cluster -o json | jq '.status.gatherStatus.gatherers[].name'"
 	// An example of disabling gatherers looks like this: `disabledGatherers: ["clusterconfig/machine_configs", "workloads/workload_info"]`
+	// +kubebuilder:validation:MaxItems=100
+	// +kubebuilder:validation:items:MaxLength=256
+	// +kubebuilder:validation:XValidation:rule=`self.all(x, x.matches("^[a-z]+[_a-z]*[a-z]([/a-z][_a-z]*)?[a-z]$"))`,message=`disabledGatherers must be in the format of {gatherer}/{function} where the gatherer and function are lowercase strings that may include underscores (_) and are separated by a forward slash (/) if the function is provided`
```

Comment:
Nit: I think there is a way to apply the validation on an individual item level using `+kubebuilder:validation:items:XValidation`

IIRC from Joel's talk last year at KubeCon NA, setting the validation on the individual items _may_ result in a lower overall cost estimate.

Marking this as a nit because I personally don't know how significant of a difference it would make and whether or not it is worth it. @JoelSpeed may have opinions here.

Reply:
Best to create a type alias and apply the validation to the type alias, then mark this as a slice of the alias

Reply:
I have created the type alias for this one